docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
+
+## Abstract
+
+A generic ``Config`` object to store command-line flags and environment variables
+that is used for configuration of ``Kubernetes`` components living on top of ``etcd``
+is proposed in this proposal.
+
+The main focus of this proposal to solve dynamic configuration problem of components 
+and encapsulate configuration information to increase overall system quality and to 
+create a flexible configuration model for ``Kubernetes`` components.
+
+## Problem
+
+Currently command-line flags and environment variables are used for component configurations in ``Kubernetes``.
+But, this approach is not suitable for a distributed computing environment. Specifically the 
+problems with the current approach are:
+
+* Hard to synchronize across replicated components.
+* Hard to change dynamically as needed by continuous services.
+* Hard to change in different versions that may support different configuration options.
+
+## Solution
+
+A new ``Config`` resource is proposed to solve the aforementioned problems 
+that is stored in ``etcd`` and distributed to interested/necessary components 
+through the API server. ``Config`` resource will be mounted as a new volume 
+and will be managed by a side-car container that communicates with the API server.
+
+The ``Config`` resource architecture will be very similar to that of ``Secrets``.
+However, the ``Config`` resource will be dynamic, meaning that different components consuming
+the same ``Config`` will be notified about the changes to that particular resource.
+
+### Advantages
+
+* Reusable across different components.
+* Easy distribution through the API server.
+* By leveraging the power of ``/watch API`` along with the new resource, we can change component configurations dynamically.
```

Comment:
Isn't dynamic vs static data orthogonal to whether the data is "secret" or "config"?
I'm fine with starting out as this suggests, but eventually, I think we might want to rationalize the two Kinds by supporting all four points combinations of (secret,config) x (dynamic, static).